homeassistant/components/tuya/sensor.py

```diff
@@ -729,6 +729,22 @@ class TuyaSensorEntityDescription(SensorEntityDescription):
             state_class=SensorStateClass.MEASUREMENT,
         ),
     ),
+    # Fan
+    # https://developer.tuya.com/en/docs/iot/s?id=K9gf48quojr54
+    "fs": (
+        TuyaSensorEntityDescription(
+            key=DPCode.COUNTDOWN_LEFT,
+            name="Remaining Time of Countdown",
+            entity_category=EntityCategory.CONFIG,
+            icon="mdi:timer-outline",
+        ),
```

Comment:
We do not allow for sensors to count down, instead, they should return a datetime object when the countdown finishes.

Reply:
I succeeded by creating a function in the `IntegerTypeData` class in `base.py`.

````
    def int_to_timestamp(
        self,
        value: float,
        unit: str | None = None,
    ) -> datetime:
        """Remap a value from this range to a new range."""
        now = dt_util.utcnow()
        if self.unit is not None:
            if unit == "ms":
                return dt_util.as_local(now) + datetime.timedelta(milliseconds=value)
            elif self.unit == "s":
                return dt_util.as_local(now) + datetime.timedelta(seconds=value)
            elif self.unit == "min":
                return dt_util.as_local(now) + datetime.timedelta(minutes=value)
````

But the sensor display is expected to look like this?
![image](https://user-images.githubusercontent.com/31328123/151393587-7d8ac622-700d-4e66-8941-2800f0374b87.png)

Reply:
Hehe no it is not, but that means the device class is missing :)

`SensorDeviceClass.TIMESTAMP`

Reply:
Got it, the problem was that even defining the `DeviceClass.TIMESTAMP` the native unit of measure got in the way.

It worked after some changes in sensor.py, please review if it is workable.

The sensor when the countdown is off will be unknown:
![image](https://user-images.githubusercontent.com/31328123/151459758-9a022b57-81bc-456a-93e9-06f4674c537a.png)

But when it's set it will work:
![image](https://user-images.githubusercontent.com/31328123/151459736-c3d58435-8840-4a8a-abe3-e746dab8e3a4.png)

Despite not being able to test on a fan, I performed actual tests with a plug. With these tests I think I understood how the countdown works on Tuya devices: it is the time to toggle the device (if it is on it will be off, if it is off it will be on / if it is open it will be closed...).